### PR TITLE
cli: Add chainlink node db migrate

### PR DESCRIPTION
**Pull request by archseer**
Intended for doing database migrations *before* starting the node. Potentially useful in k8s clusters.

Note: this now exposes `chainlink node db` outside of dev mode, but all the subcommands apart from migrate remain disabled.

### CODE REVIEW

**Review by PiotrTrzpil — COMMENTED** [review]
looks good i think

File: core/cmd/app.go [hunk 1]
```diff
@@ -627,6 +626,12 @@ func NewApp(client *Client) *cli.App {
 							Action: client.PrepareTestDatabase,
 							Flags:  []cli.Flag{},
 						},
+						{
+							Name:   "migrate",
+							Usage:  "Migrate the database.",
```
Comment (connorwstein) [inline core/cmd/app.go:631]: I'd add something about it's migrating to the latest version

File: core/cmd/app.go [hunk 2]
```diff
@@ -627,6 +626,12 @@ func NewApp(client *Client) *cli.App {
 							Action: client.PrepareTestDatabase,
 							Flags:  []cli.Flag{},
 						},
+						{
```
Comment (connorwstein) [inline core/cmd/app.go:629]: We should add a "version" command here also which shows the latest version

File: core/cmd/local_client.go [hunk 3]
```diff
@@ -381,6 +381,22 @@ func (cli *Client) PrepareTestDatabase(c *clipkg.Context) error {
...
+// MigrateDatabase migrates the database
+func (cli *Client) MigrateDatabase(c *clipkg.Context) error {
+	logger.SetLogger(cli.Config.CreateProductionLogger())
+	config := orm.NewConfig()
+	parsed := config.DatabaseURL()
+	if parsed.String() == "" {
+		return cli.errorOut(errors.New("You must set DATABASE_URL env variable. HINT: If you are running this to set up your local test database, try DATABASE_URL=postgresql://postgres@localhost:5432/chainlink_test?sslmode=disable"))
+	}
+
+	logger.Infof("Migrating database: %#v", parsed.String())
```
Comment (connorwstein) [inline core/cmd/local_client.go:393]: Would be nice to include "from migration X to migration Y" in this

File: docs/CHANGELOG.md [hunk 4]
```diff
@@ -21,6 +21,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Add `ETH_MIN_GAS_PRICE_WEI` configuration option. This defaults to 1Gwei on mainnet. Chainlink will never send a transaction at a price lower than this value.
 
+- Add `chainlink node db migrate` for running database migrations. It's
+  recommended to use this and set `MIGRATE_DATABASE=false` on more complex
```
Comment (connorwstein) [inline docs/CHANGELOG.md:25]: "on more complex deployments" is a little vague - let's just say "if you want to run the migrations separately outside of  application startup"

**Review by samsondav — DISMISSED** [review]
LGTM with connor's comments applied

File: core/cmd/app.go (hunk 2 above)
Comment (archseer) [reply core/cmd/app.go:629]: Do you mean this version? https://github.com/smartcontractkit/chainlink/blob/7af51128ef9f9e4289a2142564d0aee0420cb120/core/store/orm/orm.go#L170

It returns `random_<some long number>`

Comment (PiotrTrzpil) [reply core/cmd/app.go:629]: i think you can get the last db migration version via `getLastRunMigration` in migration_helper.go

**Review by connorwstein — APPROVED** [review]

**Pull request merged**